runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/ElementAwareAutocompleteWidget.tsx

```diff
@@ -158,25 +162,53 @@ const ElementAwareAutocompleteWidget: FC<ElementAwareAutocompleteProps> = (props
                 .filter(
                     (element: BPMNElement) =>
                         element.businessObject.actionId === 'variables.assign' ||
-                        element.businessObject.actionId === 'variables.assignList',
+                        element.businessObject.actionId === 'variables.assignList' ||
+                        element.businessObject.actionId === 'variables.assignGlobalVariable',
                 );
 
             const variables = assignVariablesElements
                 .map((assignVariablesElement) => {
                     const inputOutput: CamundaInputOutputElement = assignVariablesElement.businessObject
                         ?.extensionElements?.values[0] as CamundaInputOutputElement;
-                    if (inputOutput) {
-                        const [variable] = inputOutput.inputParameters.filter(
-                            (inputParameter) => inputParameter.name === 'variable',
-                        );
+
+                    if (!inputOutput) {
+                        return undefined;
+                    }
+                    const localVariable = inputOutput.inputParameters.find(
+                        (inputParameter) =>
+                            inputParameter.name === 'variable',
+                    );
+                    if (localVariable) {
                         return {
-                            label: variable.value,
-                            value: variable.value,
+                            label: localVariable.value,
+                            value: localVariable.value,
                             group: translate(
                                 'Process.Details.Modeler.Widgets.ElementAwareAutocomplete.Groups.Variables',
                             ),
                         };
                     }
+                    const globalVariable = inputOutput.inputParameters.find(
+                        (inputParameter) =>
+                            inputParameter.name === 'globalVariable',
+                    );
+                    if (globalVariable) {
```

Comment:
this condition should be inverted and return `undefined`. Thanks to it you have no nested ifs which implies less complexity

Reply:
also the division of the logic into sub-functions will improve readability

Reply:
I took a look at the whole `useMemo`, and it should get a minor rework. Contact me for the instructions